### PR TITLE
Fix new typos found by codespell

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
I have two commits, because the second one is debatable and changes a local variable name. I can squash it if you agree.

Definition of _fulfill_ in some dictionaries:
* [Oxford Learner's Dictionary](https://www.oxfordlearnersdictionaries.com/definition/english/fulfil)
* [Cambridge Dictionary](https://dictionary.cambridge.org/dictionary/english/fulfill)
* [Merriam-Webster](https://www.merriam-webster.com/dictionary/fulfill)
* [Collins](https://www.collinsdictionary.com/dictionary/english/fulfill)

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by paulidale — DISMISSED** [review]

File: test/cmactest.c [hunk 1]
```diff
@@ -77,7 +77,7 @@ static struct test_st {
             0x0b, 0x0c, 0x0d, 0x0e, 0x0f
         },
         16,
-        /* fullfill data to 3072 bytes with below string */
+        /* fulfill data to 3072 bytes with below string */
```
Comment (tom-cosgrove-arm) [inline test/cmactest.c:80]: `fulfill` isn't the right word, so if you're going to change it you might as well use a better word

Comment (tom-cosgrove-arm) [reply test/cmactest.c:80]: I was thinking of something a bit more descriptive, maybe

```
/* String below is repeated to give the number of bytes of data (3072) */
```

File: test/cmactest.c [hunk 2]
```diff
@@ -177,19 +177,19 @@ static int test_cmac_run(void)
 
     ctx = CMAC_CTX_new();
 
-    /* Construct input data, fullfill data to match data length */
+    /* Construct input data, fulfill data to match data length */
```
Comment (tom-cosgrove-arm) [inline test/cmactest.c:180]: This should definitely be `repeat`

Comment (DimitriPapadopoulos) [reply test/cmactest.c:180]: Then `repeat` it is, It sounds good. Ill fix that.

**Review by tom-cosgrove-arm — APPROVED** [review]
I don't think this `cmactest.c` data comments are great, but they are a lot better than they were, so I'm approving this

**Comment by tom-cosgrove-arm** [comment]
@paulidale Are you okay with the update?

**Comment by paulidale** [comment]
Yes, I'm okay with the update.

Merged, thanks for the fixes.

**Pull request closed without merging**